packages/@aws-cdk/aws-apprunner-alpha/test/service.test.ts

```diff
@@ -1350,4 +1350,544 @@ test('addToRolePolicy', () => {
+    source: apprunner.Source.fromEcrPublic({
+      imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
+    }),
+    healthCheckConfiguration: {
+      healthyThreshold: 5,
+      interval: cdk.Duration.seconds(5),
+      path: '/',
+      protocol: apprunner.HealthCheckProtocolType.HTTP,
+      timeout: cdk.Duration.seconds(2),
+      unhealthyThreshold: 5,
+    },
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::AppRunner::Service', {
+    HealthCheckConfiguration: {
+      HealthyThreshold: 5,
+      Interval: 5,
+      Path: '/',
+      Protocol: 'HTTP',
+      Timeout: 2,
+      UnhealthyThreshold: 5,
+    },
+  });
+});
+
+test('path cannot be specified with TCP protocol in healthCheckConfiguration', () => {
+  // GIVEN
+  const app = new cdk.App();
+  const stack = new cdk.Stack(app, 'demo-stack');
+
+  // WHEN
+  expect(() => {
+    new apprunner.Service(stack, 'DemoService', {
+      source: apprunner.Source.fromEcrPublic({
+        imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
+      }),
+      healthCheckConfiguration: {
+        healthyThreshold: 5,
+        interval: cdk.Duration.seconds(5),
+        path: '/',
+        protocol: apprunner.HealthCheckProtocolType.TCP,
+        timeout: cdk.Duration.seconds(2),
+        unhealthyThreshold: 5,
+      },
+    });
+  }).toThrow('path is only applicable when you set Protocol to HTTP');
+});
+
+test('path cannot be specified without HTTP protocol in healthCheckConfiguration', () => {
+  // GIVEN
+  const app = new cdk.App();
+  const stack = new cdk.Stack(app, 'demo-stack');
+
+  // WHEN
+  expect(() => {
+    new apprunner.Service(stack, 'DemoService', {
+      source: apprunner.Source.fromEcrPublic({
+        imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
+      }),
+      healthCheckConfiguration: {
+        healthyThreshold: 5,
+        interval: cdk.Duration.seconds(5),
+        path: '/',
+        timeout: cdk.Duration.seconds(2),
+        unhealthyThreshold: 5,
+      },
+    });
+  }).toThrow('path is only applicable when you set Protocol to HTTP');
+});
+
+test('path cannot be empty in healthCheckConfiguration', () => {
+  // GIVEN
+  const app = new cdk.App();
+  const stack = new cdk.Stack(app, 'demo-stack');
+
+  // WHEN
+  expect(() => {
+    new apprunner.Service(stack, 'DemoService', {
+      source: apprunner.Source.fromEcrPublic({
+        imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
+      }),
+      healthCheckConfiguration: {
+        healthyThreshold: 5,
+        interval: cdk.Duration.seconds(5),
+        path: '',
+        protocol: apprunner.HealthCheckProtocolType.HTTP,
+        timeout: cdk.Duration.seconds(2),
+        unhealthyThreshold: 5,
+      },
+    });
+  }).toThrow('path length must be greater than 0');
+});
+
+test('OK if healthyThreshold is 1 in healthCheckConfiguration', () => {
```

Comment:
I think that we should only leave the unit tests for the failing cases of validation, we are already checking that the implementation is working in the `Service has healthCheckConfiguration` test.
Adding test cases for checking limit values is a bit redundant in my opinion.